app/controllers/snippets_controller.rb

```diff
@@ -6,8 +6,11 @@ def new
   def create
     @snippet = Snippet.new(snippet_params)
     @snippet.user_id = current_user.id
-    @snippet.save
-    redirect_to dashboard_path
+    if @snippet.save
+      redirect_to dashboard_path
+    else
+      redirect_to new_snippet_path, notice: "Please fill out all fields."
```

Comment:
How about using `@snippet.errors.messages`? Not sure if that's what you had in mind or if there's a better way.